slatedb/src/admin.rs

```diff
@@ -127,34 +120,24 @@ impl Admin {
         gc_opts: GarbageCollectorOptions,
         cancellation_token: CancellationToken,
     ) -> Result<(), Box<dyn Error>> {
-        let manifest_store = Arc::new(ManifestStore::new(&self.path, self.object_store.clone()));
-        manifest_store
-            .validate_no_wal_object_store_configured()
```

Comment:
@taburet Please note I'm removing this check. I believe things will work properly now. The AdminBuilder and GarbageCollectorBuilder both allow `with_wal_object_store`. Consequently, the GC should properly collect WAL files on a different object store.

LMk if you have any concerns.